var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -423,9 +423,17 @@ def common_config_args(self):
             ])
             if self.spec.satisfies('@:5.11'):
                 config_args.append('-no-xinput2')
-
         # FIXME: else: -system-xcb ?
 
+        platform = None
+        if MACOS_VERSION:
+            platform = 'unsupported/macx-clang-libc++'
```

Comment:
Will this work for non-Clang compilers? What about Apple Clang vs LLVM Clang on macOS?

Reply:
The patches that I needed to get QT4 to work on mac are only for apple-clang. I'm not sure what would be necessary to get QT to compile on other Mac compilers with any of the QT versions: so leaving the line as above was to my mind the safest way to avoid breaking the existing `qt@4% clang@apple platform=darwin` configuration 😬

Reply:
It sounds like this won't work for `%gcc platform=darwin`. I would use `self.spec.satisfies('%clang platform=darwin')`.

Reply:
Good point, I've changed the code to try to support QT4 and QT5 for linux and darwin with or without clang.